src/db.ts

```diff
@@ -129,18 +134,26 @@ export class NostroidDexie extends Dexie {
           p.nip05Valid = null
         }
       }
+
       db.profiles.bulkPut(profiles)
-    })
-    this.validateNip05ProfilesNip05(profiles || [])
+
+      return profiles;
+    }).then((profiles) => {this.validateNip05ProfilesNip05(profiles ?? [])});
```

Comment:
why mix await with promise.then?

Why not:

profiles = await db.transaction ...
validate(profiles)